test/csuite/wt5212_incremental_backup/main.c

```diff
@@ -0,0 +1,659 @@
+                if (strncmp(filename, WTLOG, WTLOGLEN) == 0)
+                    testutil_check(__wt_snprintf(buf, sizeof(buf), "cp %s/%s/%s %s/%s/%s", home,
+                      logpath, filename, h, logpath, filename));
+                else
+                    testutil_check(__wt_snprintf(
+                      buf, sizeof(buf), "cp %s/%s %s/%s", home, filename, h, filename));
+#if 0
+                printf("Incremental: Whole file copy: %s\n", buf);
+#endif
+                testutil_check(system(buf));
+            }
+        }
+        scan_end_check(ret == WT_NOTFOUND);
+        /* Done processing this file. Close incremental cursor. */
+        testutil_check(incr_cur->close(incr_cur));
+
+        /* Close file descriptors if they're open. */
+        if (rfd != -1) {
+            testutil_check(close(rfd));
+            testutil_check(close(wfd));
+            // Reset file descriptors to default value after closing.
+            rfd = wfd = -1;
+        }
+        /*
+         * For each file, we want to copy the file into each of the later incremental directories so
+         * that they start out at the same for the next incremental round. We then check each
+         * incremental directory along the way.
+         */
+        for (j = counter; j <= MAX_ITERATIONS; j++) {
+            testutil_check(__wt_snprintf(h, sizeof(h), "%s.%d", home_incr, j));
+            if (strncmp(filename, WTLOG, WTLOGLEN) == 0)
+                testutil_check(__wt_snprintf(buf, sizeof(buf), "cp %s/%s/%s %s/%s/%s", home,
+                  logpath, filename, h, logpath, filename));
+            else
+                testutil_check(
+                  __wt_snprintf(buf, sizeof(buf), "cp %s/%s %s/%s", home, filename, h, filename));
+            testutil_check(system(buf));
+        }
+    }
+    scan_end_check(ret == WT_NOTFOUND);
+
+    /* Done processing all files. Close backup cursor. */
+    testutil_check(backup_cur->close(backup_cur));
+    finalize_files(flist, count);
+    free(tmp);
+}
+
+/*
+ * This function will add records to the table (table:main), take incremental/full backups and
+ * validate the backups.
+ */
+static void
+add_data_validate_backups(WT_SESSION *session)
+{
+    printf("Adding initial data\n");
+    /*
+     * We set initial_backup = true to take a full backup into the incremental directories.
+     */
+    initial_backup = true;
+    add_work(session, uri, false);
+    printf("Taking initial backup\n");
+    take_full_backup(session);
+    initial_backup = false;
+    testutil_check(session->checkpoint(session, NULL));
+
+    add_work(session, uri, false);
+    take_full_backup(session);
+    take_incr_backup(session);
+    testutil_check(compare_backups(uri));
+}
+
+/*
+ * This function will remove all the records from table (table:main), take backup and validate the
+ * backup.
+ */
+static void
+remove_all_records_validate(WT_SESSION *session)
+{
+    remove_work(session);
+    take_full_backup(session);
```

Comment:
I think the intent for all these cases is that we don't need to take the full backup, but we do the operation and take only an incremental.